File: 1D Array/java/program7.java
import java.util.Scanner;

public class program7 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements in the array: ");
        int n = scanner.nextInt();

        if (n < 2) {
            System.out.println("Second largest number doesn't exist");
            scanner.close();
            return;
        }

        int[] arr = new int[n];
        System.out.print("Enter the element in arr[0]: ");
        arr[0] = scanner.nextInt();
        System.out.print("Enter the element in arr[1]: ");
        arr[1] = scanner.nextInt();

        int max, smax;

        if (arr[0] > arr[1]) {
            max = arr[0];
            smax = arr[1];
        } else {
            max = arr[1];
            smax = arr[0];
        }

        for (int i = 2; i < n; i++) {
            System.out.print("Enter the element in arr[" + i + "]: ");
            arr[i] = scanner.nextInt();

            if (arr[i] > max) {
                smax = max;
                max = arr[i];
            } else if (arr[i] > smax && arr[i] < max) {
                smax = arr[i];
            }
        }

        if (smax == max) {
            System.out.println("There is no second max element in this array");
        } else {
            System.out.println("The second largest element in this array is: " + smax);
        }

        scanner.close();
    }
}

File: 1D Array/C/program_7.c
/* Find Second largest element in an array
 Given an array of integers, our task is to write a program 
that efficiently finds the second-largest element present in 
the array. 
Examples:
 Input: arr[] = {12, 35, 1, 10, 34, 1}
 Output: The second largest element is 34.
 Explanation: The largest element of the array is 35 and the 
second largest element is 34
 Input: arr[] = {10, 5, 10}
 Output: The second largest element is 5.
 Explanation: The largest element of the array is 10 and the 
second largest element is 5*/
#include<stdio.h>
int main(){
    int n;
    printf("enter the number of the element in the array :");
    scanf("%d",&n);
    if(n<2){
        printf("second largest number doesn't exist");
        return 0;
    }
    int arr[n];

    printf("enter the element in the arr[0]:");
    scanf("%d",&arr[0]);

    printf("enter the element in the arr[1]:");
    scanf("%d",&arr[1]);

     int max;
     int smax;

    if(arr[0] > max){
        max = arr[0];
        smax = arr[1];
    }
    else{
        max = arr[1];
        smax = arr[0];
    }

   
    for(int i=2;i<n;i++){
        printf("enter the elements in the arr[%d]:",i);
        scanf("%d",&arr[i]);

        if(arr[i]>max){
            smax = max;
            max = arr[i];
        }
        else if(arr[i] > smax && arr[i] < max){
            smax =arr[i];

        }

    }

    if(smax == max){
        printf("there is no second max element exist in this array :");
    }
    else{
        printf("the second largest element in this array is :%d",smax);
    }

    return 0;
}